### ISSUE TITLE
Does every List type need to be a Connection?

### ISSUE THREAD

**Issue by boosh** [facebook/relay#171]
I'm trying to return a list of event items. My server correctly returns the following:
```
{"data":{"events":[{"id":"8","title":"Click 1"},{"id":"20","title":"View 2"}]}}
```
My component is only being provided with the first item in the list though, i.e. the event with ID 8, instead of the whole list. Here's my component:
```
class Event extends React.Component {
  render() {
    var this.props.events.map(...); // error because events is not a list but it should be
    return (
        <div>
          Title: {this.props.events.title}         // works but shouldn't
        </div>
    );
  }
}

export default Relay.createContainer(Event, {
  fragments: {
    events: () => Relay.QL`
      fragment on Event {
        id,
        title
      }
    `,
  },
});
```
And the route:
```
export default class extends Relay.Route {

  static queries = {
    events: Component => Relay.QL`
      query {
        events(id: $userID) {
          ${Component.getFragment('events')},
        },
      }
    `,
  };

  static paramDefinitions = {
    userID: {required: true},
  };

  static routeName = 'EventRoute';
}
```
Looking at the examples (todo, the one in relay-starter-kit) I can see that all of the list types are wrapped in Connection objects that provide metadata. Is this a requirement? I'd have expected the above to work and to be able to extend it in future by changing it to a Connection object when I needed to, instead of having to do it to begin with.

**Comment by yuzhi** [comment]
Not all list type are required to be in a Connection. If you don't need any pagination logic, it's sufficient to just have the field return an array of events `[Event]`.

I noticed a few things that might cause problems in your current setup:
- Relay currently doesn't support plural root fields, see #112 for more information and workaround. We are looking into better support for root fields
- If you are planning on using the `Event` component to render all the events in the props, you have to specify in the fragment to be plural with `fragment on Event @relay(plural: true)`

**Comment by boosh** [comment]
I tried specifying the fragment to be plural and I got an error: 
```
Invariant Violation: GraphQLFragmentPointer: Wrong plurality, single data ID supplied with plural fragment.
```
I only want to give a single user ID and get a list of events back for them. I guess I don't need to make it plural. I still can't get it working. This is a root field with one argument that queries a single node.

My server returns multiple Events, but relay is only picking up the first one for some reason. Any other ideas what might be causing this? I don't need pagination, etc. so don't need a Connection...
